beacon_node/beacon_chain/src/light_client_server_cache.rs

```diff
@@ -127,13 +135,13 @@ impl<T: BeaconChainTypes> LightClientServerCache<T> {
         if is_latest_finality & !cached_parts.finalized_block_root.is_zero() {
             // Immediately after checkpoint sync the finalized block may not be available yet.
             if let Some(finalized_block) =
-                store.get_blinded_block(&cached_parts.finalized_block_root)?
+                store.get_full_block(&cached_parts.finalized_block_root)?
             {
                 *self.latest_finality_update.write() = Some(LightClientFinalityUpdate {
                     // TODO: may want to cache this result from latest_optimistic_update if producing a
                     // light_client header becomes expensive
-                    attested_header: block_to_light_client_header(attested_block.message()),
-                    finalized_header: block_to_light_client_header(finalized_block.message()),
+                    attested_header: block_to_light_client_header(fork_name, attested_block)?,
+                    finalized_header: block_to_light_client_header(fork_name, finalized_block)?,
```

Comment:
For both the `attested_header` and the `finalized_header` I'm getting the `fork_name` from the `attested_block`. This might be a stupid question, but I'm wondering if there could be fork boundary issues here where the `attested_block` is in a different fork than the `finalized_block`?

Reply:
> This might be a stupid question, but I'm wondering if there could be fork boundary issues here where the attested_block is in a different fork than the finalized_block?

Definitely not a stupid question. We try to schedule forks such that the fork epoch is a multiple of 256, to keep historical vectors in the same fork. Indirectly that makes lightclient periods align with fork boundaries. However some other network, testnet etc could choose a fork boundary such that it does not align with the start of the sync committee period. For simplicity we can accept to crash for now.

Reply:
I've thought about this before as well and forgot to ask someone. Thanks for this question @eserilev and thanks for the asnwer @dapplion